### PR TITLE
feat(telephony.alias): redesign of ccs and svi config main page

### PR DESCRIPTION
**Pull request by JeremyDec**
## Redesign of Alias/Configuration main page for contact center solution (and expert) and svi (and expert) aliases
MANAGER-977
MANAGER-978

### Description of the Change
Replacement of old page "Configuration du numéro" by a new freshly one.

(Metro plan is now used only by 3 features type : svi, cloudIvr and ccs expert).

Members, Mode, Calls filtering and Time condition pages are now directly accessible, for one type of aliases.

LOTS of deleted files !! 

So don't be scared by the amount of changes files :-)
Some other ones have just the controller be renamed, no change in the html structure, it will be done later. (by later feature)

### CODE REVIEW

File: client/app/telecom/telephony/alias/configuration/callsFiltering/telecom-telephony-alias-configuration-callsFiltering.html [hunk 1]
```diff
@@ -7,12 +7,12 @@
         <h1 data-translate="telephony_alias_configuration_calls_filtering_title"></h1>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/callsFiltering/telecom-telephony-alias-configuration-callsFiltering.html]: This whole file will be rewrite in a later feature, I've just changed the controller name.

File: client/app/telecom/telephony/alias/configuration/members/telecom-telephony-alias-configuration-members.html [hunk 2]
```diff
@@ -7,12 +7,12 @@
         <h1 data-translate="telephony_alias_configuration_members_title"></h1>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/members/telecom-telephony-alias-configuration-members.html]: This whole file will be rewrite in a later feature, I've just changed the controller name.

File: client/app/telecom/telephony/alias/configuration/mode/telecom-telephony-alias-configuration-mode.html [hunk 3]
```diff
@@ -7,12 +7,12 @@
         <h1 data-translate="telephony_alias_configuration_mode_title"></h1>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/mode/telecom-telephony-alias-configuration-mode.html]: This whole file will be rewrite in a later feature, I've just changed the controller name.

File: client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js [hunk 4]
```diff
@@ -5,8 +5,25 @@ angular.module('managerApp').config(($stateProvider) => {
       'aliasInnerView@telecom.telephony.alias': {
         templateUrl: 'app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.html',
         controller: 'TelecomTelephonyAliasConfigurationCtrl',
-        controllerAs: 'AliasConfigurationCtrl',
+        controllerAs: '$ctrl',
       },
+      'featureView@telecom.telephony.alias.configuration': {
+        templateProvider: alias => `<div data-ng-include="::'app/telecom/telephony/alias/configuration/feature/${alias.featureType}/telecom-telephony-alias-configuration-${alias.featureType}.html'"></div>`,
+        controllerProvider: (alias) => {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js]: I know I've could destructure the param, but for readability, I've let the param this way

File: client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js [hunk 5]
```diff
@@ -5,8 +5,25 @@ angular.module('managerApp').config(($stateProvider) => {
...
+          return isNotEmptyFeature ? `TelecomTelephonyAliasConfiguration${_.capitalize(alias.featureType)}Ctrl` : null;
+        },
+        controllerAs: '$ctrl',
+      },
+    },
+    resolve: {
+      alias: ($stateParams, voipService) => voipService
+        .fetchSingleService($stateParams.billingAccount, $stateParams.serviceName)
+        .then((alias) => {
+          const _alias = _.assign(alias);
```
Comment (marie-j) [inline client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js]:
```suggestion
          const _alias = _.clone(alias);
```

Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js]: Done through commit amended :)

File: client/app/telecom/telephony/alias/configuration/feature/contactCenterSolution/telecom-telephony-alias-configuration-contactCenterSolution.controller.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    this.$uibModal.open({
+      templateUrl: 'app/telecom/telephony/alias/dashboard/confirmDeleteConfiguration/confirmDeleteConfiguration.modal.html',
+      controller: 'TelecomTelephonyAliasConfirmDeleteConfigurationCtrl',
+      controllerAs: '$ctrl',
+      backdrop: 'static',
+      resolve: {
+        number: this.alias,
+      },
+    }).result.then(() => {
+      this.OvhApiTelephony.Service().v6().resetCache();
```
[the remark this answers is not in the document]
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/feature/contactCenterSolution/telecom-telephony-alias-configuration-contactCenterSolution.controller.js]: why to return this ? This isn't a promise

[the remark this answers is not in the document]
Comment (FredericEspiau) [reply client/app/telecom/telephony/alias/configuration/feature/contactCenterSolution/telecom-telephony-alias-configuration-contactCenterSolution.controller.js]: I think I was just trying the new `suggestion` feature 😄

File: client/components/telecom/voip/service/alias/voip-service-alias.factory.js [hunk 7]
```diff
@@ -33,6 +33,21 @@ angular.module('managerApp').factory('VoipServiceAlias', (VoipService) => {
...
+     *  @name managerApp.object:VoipServiceAlias#isContactCenterSolution
+     *  @propertyOf managerApp.object:VoipServiceAlias
+     *
+     *  @description
+     *  Check if the alias feature type is equivalent to Contact Center Solution feature
+     *
+     *  @return {Boolean}
+     */
+    isContactCenterSolution() {
+      const ccsGroupedFeatures = ['contactCenterSolution', 'easyHunting', 'cloudHunting', 'easyPabx', 'miniPabx', 'oldPabx', 'ovhPabx'];
```
[the remark this answers is not in the document]
Comment (antleblanc) [reply client/components/telecom/voip/service/alias/voip-service-alias.factory.js]: Does it make sense to have them as a constant?

File: client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.html [hunk 8]
```diff
@@ -1,21 +1,35 @@
...
+        <div class="col-lg-8">
+            <oui-collapsible class="mb-5"
+                             data-heading="{{:: 'telephony_alias_configuration_question_' + $ctrl.groupFeatureType() | translate }}">
+                <div data-ng-if="$ctrl.number.feature.featureType !== 'redirect'">
+                    <p class="mb-4"
+                       data-translate="{{:: 'telephony_alias_config_feature_type_' + $ctrl.groupFeatureType() + '_first_part' }}"
+                       data-translate-values="{ 'featureType': '<strong>' + $ctrl.featureTypeLabel + '</strong>' }"></p>
+                    <p data-translate="{{:: 'telephony_alias_config_feature_type_' + $ctrl.groupFeatureType() + '_second_part' }}"></p>
+                </div>
+                <div data-ng-if="$ctrl.number.feature.featureType === 'redirect'">
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.html]: You can apply `ngIf` directly on the `<span>` element.

File: client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js [hunk 9]
```diff
@@ -5,8 +5,25 @@ angular.module('managerApp').config(($stateProvider) => {
...
+          return isNotEmptyFeature ? `TelecomTelephonyAliasConfiguration${_.capitalize(alias.featureType)}Ctrl` : null;
+        },
+        controllerAs: '$ctrl',
+      },
+    },
+    resolve: {
+      alias: ($stateParams, voipService) => voipService
+        .fetchSingleService($stateParams.billingAccount, $stateParams.serviceName)
+        .then((alias) => {
+          const _alias = _.clone(alias);
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js]: Even if the ESLint rule `no-underscore-dangle` is disabled, coud we please respect this rule?

File: client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml [hunk 10]
```diff
@@ -38,4 +38,59 @@
...
+
+   <translation id="telephony_alias_config_change_type_label_cloudHunting" qtlid="333165">File d'appels (mode expert)</translation>
+   <translation id="telephony_alias_config_change_type_desc_cloudHunting">Avec le mode expert de la file d’appels, vous pouvez gérer le flux de vos appels de bout en bout.<br/>Sur une interface similaire à celle du serveur vocal interactif, programmez étape par étape le déroulement de vos appels entrants.<br/>En fonction des numéros appelants ou de plages horaires, jouez des sons d’accueils, personnalisez votre musique d’attente, faites sonner vos collaborateurs ou différents services suivant des stratégies que vous définissez.<br/>Notez que dans ce mode de fonctionnement, les renvois d’appels des membres de votre file d’appels seront automatiquement désactivés.</translation>
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml]: Could you split into multiple `<translation>` keys? (avoid `<br/>` tag I mean)

File: client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js (hunk 9 above)
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/telecom-telephony-alias-configuration.js]: Ok for sure :)

File: client/components/telecom/voip/service/alias/voip-service-alias.factory.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (JeremyDec) [reply client/components/telecom/voip/service/alias/voip-service-alias.factory.js]: I've just used a constant with a name to show that those features are grouped

File: client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml (hunk 10 above)
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml]: In fact, this one can be deleted

**Pull request merged**